### ISSUE TITLE
OpenAPI planner agent doesn't support large specs

### ISSUE THREAD

**Issue by voluntadpear** [langchain-ai/langchain#2786]
When replicating the [hierarchical planning example](https://python.langchain.com/en/latest/modules/agents/toolkits/examples/openapi.html#st-example-hierarchical-planning-agent) with a large enough OpenAPI specification, the following error is thrown when running the agent with any query:

```
InvalidRequestError: This model's maximum context length is 4097 tokens. However, your messages resulted in 6561 tokens. Please reduce the length of the messages.
```

Here is how I'm reducing my OpenAPI spec:

```python
with open("server.yml") as f:
    raw_server_spec = yaml.load(f, Loader=yaml.Loader)
server_spec = reduce_openapi_spec(raw_server_spec)
```

And here's how I'm initializing the agent:

```python
llm = ChatOpenAI(temperature=0.0)
openapi_agent = create_planner_openapi_agent(server_spec, requests_wrapper, llm)
user_query = "Return the response for retrieving document info for the document with id 1"
openapi_agent.run(user_query);
```

I think the OpenAPI spec reducer should have a way of splitting the spec into multiple chunks if necessary and the OpenAPI agent adapted to go across many chunks if needed, perhaps with a map-reduce or "stuff" approach.

**Comment by RizziGit93** [comment]
Hey Guillermo! Im facing a problem like yours. Im trying the same example with a huge OpenAPI spec. It needs 23k tokens and Im still looking for a solution. Did you find a way yo do that?

**Comment by borisko123** [comment]
Also interesting in solution

**Comment by s04** [comment]
Small update on this. OpenAI released a larger context gpt-3.5-turbo model with 16k context.

`llm = ChatOpenAI(model="gpt-3.5-turbo-16k",openai_api_key=openai_api_key,temperature=0.0)`
